src/error.ts

```diff
@@ -1,8 +1,51 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
+import { DecomposedJwt } from "./jwt.js";
+
+/**
+ * Base Error for all other errors in this file
+ */
 export abstract class JwtBaseError extends Error {}
 
+/**
+ * Interface for an error that is raised because an actual value does not match with the expected value
+ */
+interface AssertionError extends JwtBaseError {
+  failedAssertion: {
+    actual: unknown;
+    expected?: string | string[];
```

Comment:
Can this be an JSON? What is the assumption here, maybe is worth to mention.

Reply:
Sorry mate what do you mean exactly?